### PR TITLE
Camera following

### LINKED ISSUE #8: Add rooms and camera following
Add rooms like in TBOI, camera should follow the player inside the room (excluding room bounds).
When player moves through the bridge camera should follow him/room should move like in TBOI ?

### PR DESCRIPTION
**Pull request by MAKMED1337**
Add simple logic for the camera following and add some basic rooms

### CODE REVIEW

File: core/src/com/po/fuck/Camera.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.po.fuck.updates.Updatable;
+import com.badlogic.gdx.utils.ScreenUtils;
+
+import static com.po.fuck.Constants.GAME_HEIGHT;
+import static com.po.fuck.Constants.GAME_WIDTH;
+import static com.po.fuck.Constants.CAMERA_SPEED;
+
+public final class Camera extends CenterDrawer implements Updatable {
+    OrthographicCamera camera = new OrthographicCamera();
+    private Vector2 position = new Vector2();
```
Comment (welleyth) [inline core/src/com/po/fuck/Camera.java:17]: Please add a comment explaining the meaning of position. Is it a coordinate of Left-right corner, centre, or something else?

File: core/src/com/po/fuck/FUCK.java [hunk 2]
```diff
@@ -1,67 +1,62 @@
...
 
-        batch.begin();
-        All.drawableCollection.draw(new CenterDrawer(batch));
-        batch.end();
-    }
-
-    @Override
-    public void dispose() {
-        batch.dispose();
+        camera.get().draw_all();
```
Comment (welleyth) [inline core/src/com/po/fuck/FUCK.java:51]: What?

File: core/src/com/po/fuck/KeyboardController.java [hunk 3]
```diff
@@ -4,58 +4,78 @@
...
+        if (W)
+            result.y -= 1;
+        if (A)
+            result.x -= 1;
+        if (S)
+            result.y += 1;
+        if (D)
+            result.x += 1;
+        return result;
+    }
```
Comment (welleyth) [inline core/src/com/po/fuck/KeyboardController.java:31]: Should we scale result Vector2 to the length of `speed`?

File: core/src/com/po/fuck/MouseController.java [hunk 4]
```diff
@@ -50,10 +49,15 @@ public boolean touchUp(int screenX, int screenY, int pointer, int button) {
...
+        Vector2 actualPosition = centerPosition.cpy().add(FUCK.camera.get().getPosition());
+
+        Weapon weapon = getWeapon();
+        if (weapon == null)
+            return;
+
+        weapon.aim(actualPosition);
+        if (pressed)
+            weapon.attack();
     }
```
Comment (welleyth) [inline core/src/com/po/fuck/MouseController.java:68]: I don't get what happens here in code. Comments would be appreciated.

File: core/src/com/po/fuck/Room.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                start();
+            return;
+        } else if (state == State.FIGHT) {
+            if (!inside.stream().anyMatch(x -> x instanceof Entity && ((Entity) x).getTeamTag() == ENEMY_TEAM_TAG))
+                clear();
+            return;
+        } else
+            throw new AssertionError("Unknown state");
+    }
+}
```
Comment (welleyth) [inline core/src/com/po/fuck/Room.java:137]: I really would appreciate more comments

**Review by RyhorKoshman — APPROVED** [review]

File: core/src/com/po/fuck/Camera.java (hunk 1 above)
Comment (MAKMED1337) [reply core/src/com/po/fuck/Camera.java:17]: Every position in our code is the center, why should this position differ ?

File: core/src/com/po/fuck/FUCK.java (hunk 2 above)
Comment (MAKMED1337) [reply core/src/com/po/fuck/FUCK.java:51]: What is wrong ? Draw is not in dispose if you are thinking about it, look for the whole code

File: core/src/com/po/fuck/KeyboardController.java (hunk 3 above)
Comment (MAKMED1337) [reply core/src/com/po/fuck/KeyboardController.java:31]: No, it is only calculating direction

File: core/src/com/po/fuck/FUCK.java (hunk 2 above)
Comment (welleyth) [reply core/src/com/po/fuck/FUCK.java:51]: Stupid me and misleading github

**Review by welleyth — APPROVED** [review]
LGTM

**Review by RyhorKoshman — APPROVED** [review]

**Pull request merged**